common/src/KokkosKernels_SimpleUtils.hpp

```diff
@@ -380,6 +380,38 @@ KOKKOS_FORCEINLINE_FUNCTION Value xorshiftHash(Value v) {
              : static_cast<Value>(x * 2685821657736338717ULL - 1);
 }
 
+struct ViewHashFunctor {
+  ViewHashFunctor(const uint8_t *data_) : data(data_) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(size_t i, uint32_t &lhash) const {
+    // Compute a hash/digest of both the index i, and data[i]. Then add that to
+    // overall hash.
+    uint32_t x = uint32_t(i);
+    x ^= x << 13;
+    x ^= x >> 17;
+    x ^= x << 5;
+    x ^= uint32_t(data[i]);
+    x ^= x << 13;
+    x ^= x >> 17;
+    x ^= x << 5;
+    lhash += x;
+  }
+
+  const uint8_t *data;
+};
+
+// Compute a hash of a contiguous view's data.
```

Comment:
The intent of this function is to return a unique hash based on the view's data? In short, two views of the same type and same data should return the same hash? What if two views contain the same data but the element order is different?

It would be helpful to expand this function's documentation in the form of a doxygen-style comment.

Reply:
Yes, I can document this better. I think it might be useful in other places too.

Yes, because the index i is part of the digest of each element (and not in an associative way), swapping two elements in a view is almost guaranteed to change the hash (roughly 1 in 2^32 chance to collide)

Reply:
The comments for hashView are now valid Doxygen.